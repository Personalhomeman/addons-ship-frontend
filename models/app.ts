export type App = {
  appSlug: string;
  plan: string;
  title: string;
  avatarUrl: string;
  projectType: ProjectType;
  androidErrors: string[];
  iosErrors: string[];
  colors?: {
    start?: string;
    end?: string;
  };
};

export type ProjectType =
  | 'xamarin'
  | 'ios'
  | 'osx'
  | 'macos'
  | 'android'
  | 'cordova'
  | 'ionic'
  | 'react-native'
  | 'flutter'
  | 'other';
