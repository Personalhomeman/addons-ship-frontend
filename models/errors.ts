export class RequestError extends Error {
  constructor(public status: number, public statusText: string) {
    super(`${status}: ${statusText}`);
  }
}
