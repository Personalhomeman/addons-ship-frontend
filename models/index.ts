import { NextContext } from 'next';
import { Store } from 'redux';
import { Request } from 'express';

export type AppVersionProps = {
  appSlug: string;
  versionId: string;
};

export const AppVersionPageTabs = ['details', 'devices', 'test', 'activity'];

export type AppVersionPageQuery = AppVersionProps & {
  isPublic?: string;
  selectedTab?: typeof AppVersionPageTabs[number];
};

export type AppPageQuery = {
  appSlug: string;
};

export type AppVersionList = Array<{
  version: string;
  appVersions: AppVersion[];
}>;

export type AppVersion = {
  id: number;
  version: string;
  platform: string;
  buildNumber: number;
  buildSlug: string;
  lastUpdate: Date;
  description: string;
  whatsNew: string;
  minimumOs: string;
  minimumSdk: string;
  packageName: string;
  certificateExpires: Date;
  bundleId: string;
  size: number;
  supportedDeviceTypes: string[];
  distributionType: string;
  iconUrl: string;
  appName: string;
  promotionalText: string;
  keywords: string;
  reviewNotes: string;
  supportUrl: string;
  marketingUrl: string;
  scheme: string;
  configuration: string;
  publicInstallPageURL: string;
};

export interface PageContext extends NextContext {
  store: Store;
  isServer: boolean;
  req: Request;
}

export type Platform = 'ios' | 'android';
