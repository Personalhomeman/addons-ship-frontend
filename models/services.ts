export type APIConfig = {
  url: string;
};

export type AppConfig = {
  baseUrl: string;
};
