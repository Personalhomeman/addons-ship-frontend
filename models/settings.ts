export const AppSettingsPageTabs = ['general', 'notifications'];

export type AppSettingsPageQuery = {
  appSlug: string;
  selectedTab?: typeof AppSettingsPageTabs[number];
};

export type IosSettings = {
  artifactExposingWorkflows: string;
  appleDeveloperAccountEmail: string;
  appSku: string;
  appSpecificPassword: string;
  selectedProvProfile: any;
  selectedCertificate: any;
};

export type AndroidSettings = {
  artifactExposingWorkflows: string;
  track: string;
  selectedKeystoreFile: any;
  selectedServiceAccountJsonFile: any;
};

export type ProvProfile = { name: string };
export type Certificate = { name: string };
export type KeystoreFile = { name: string };
export type ServiceAccountJsonFile = { name: string };

export type Settings = {
  iosSettings?: IosSettings;
  androidSettings?: AndroidSettings;
  provProfiles?: ProvProfile[];
  certificates?: Certificate[];
  keystoreFiles?: KeystoreFile[];
  serviceAccountJsonFiles?: ServiceAccountJsonFile[];
  projectType:
    | 'xamarin'
    | 'ios'
    | 'osx'
    | 'macos'
    | 'android'
    | 'cordova'
    | 'ionic'
    | 'react-native'
    | 'flutter'
    | 'other';
};

export type AppContact = {
  id: string;
  email: string;
  isConfirmed: boolean;
  notificationPreferences: {
    newVersion: boolean;
    successfulPublish: boolean;
    failedPublish: boolean;
  };
  isMarkedForDelete?: boolean;
  isMarkedForUpdate?: boolean;
};
