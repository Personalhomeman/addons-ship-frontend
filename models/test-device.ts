export type TestDevice = {
  deviceId: string;
  deviceType: string;
  owner: string;
};
