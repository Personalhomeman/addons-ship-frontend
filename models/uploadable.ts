export type Uploadable = {
  filename: string;
  filesize: number;
  deviceType: string;
  screenSize: string;
  uploadUrl?: string;
};
