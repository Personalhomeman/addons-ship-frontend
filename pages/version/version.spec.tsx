import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import { AppVersion } from '@/models';
import { VersionPage } from './';

const mockAppVersion: AppVersion = {
  id: 123,
  version: '1.0.3',
  platform: 'ios',
  buildNumber: 32,
  buildSlug: '13245ads',
  lastUpdate: new Date('2018-09-22T12:42:31Z'),
  description: 'Some semi-long description',
  whatsNew: 'Some longer whats-new',
  minimumOs: '10.2',
  minimumSdk: 'dunno',
  packageName: 'com.package.name',
  certificateExpires: new Date('2018-09-22T12:42:31Z'),
  bundleId: 'bundle-id',
  size: 4567034,
  supportedDeviceTypes: ['iphone', 'ipad'],
  distributionType: 'development',
  iconUrl: 'http://placekitten.com/160/160',
  appName: 'Standup Timer',
  promotionalText: 'Promotional Text',
  keywords: 'Keywords',
  reviewNotes: 'Review Notes',
  supportUrl: 'https://bitrise.io/support',
  marketingUrl: 'https://bitrise.io/marketing',
  scheme: 'piramid',
  configuration: 'canary'
};

it('renders correctly', () => {
  const tree = toJSON(shallow(<VersionPage appVersion={mockAppVersion} appSlug="some-app" versionId="a-version-id" />));
  expect(tree).toMatchSnapshot();
});
