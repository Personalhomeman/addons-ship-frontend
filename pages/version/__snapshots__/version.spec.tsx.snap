// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppVersion dispatches the proper actions calculates path correctly on the client too 1`] = `
Object {
  "appSlug": undefined,
  "isPublic": undefined,
  "pagePath": "/such/path",
  "selectedTab": "details",
  "versionId": undefined,
}
`;

exports[`AppVersion dispatches the proper actions dispatches fetchAppVersion and fetchSettings 1`] = `
Object {
  "appSlug": undefined,
  "isPublic": undefined,
  "pagePath": "some/path",
  "selectedTab": "details",
  "versionId": undefined,
}
`;

exports[`AppVersion dispatches the proper actions dispatches fetchAppVersionEvents 1`] = `
Object {
  "appSlug": undefined,
  "isPublic": undefined,
  "pagePath": "some/path",
  "selectedTab": "activity",
  "versionId": undefined,
}
`;

exports[`AppVersion dispatches the proper actions dispatches fetchTestDevices 1`] = `
Object {
  "appSlug": undefined,
  "isPublic": undefined,
  "pagePath": "some/path",
  "selectedTab": "devices",
  "versionId": undefined,
}
`;

exports[`AppVersion renders the activity tab correctly 1`] = `
<Base
  paddingVertical="x10"
>
  <Base
    maxWidth={960}
  >
    <Tabs
      gap="x10"
    >
      <Tab
        active={false}
      >
        <Link
          as="some/path/details"
          href="/version?appSlug=app-slug-123&versionId=123&selectedTab=details"
        >
          <a>
            Details
          </a>
        </Link>
      </Tab>
      <Tab
        active={false}
      >
        <Link
          as="some/path/devices"
          href="/version?appSlug=app-slug-123&versionId=123&selectedTab=devices"
        >
          <a>
            Devices
          </a>
        </Link>
      </Tab>
      <Tab
        active={false}
      >
        <Link
          as="some/path/qa"
          href="/version?appSlug=app-slug-123&versionId=123&selectedTab=qa"
        >
          <a>
            Qa
          </a>
        </Link>
      </Tab>
      <Tab
        active={true}
      >
        <Link
          as="some/path/activity"
          href="/version?appSlug=app-slug-123&versionId=123&selectedTab=activity"
        >
          <a>
            Activity
          </a>
        </Link>
      </Tab>
    </Tabs>
  </Base>
  <Divider
    color="gray-2"
    direction="horizontal"
    width="0.125rem"
  />
  <Base
    maxWidth={960}
  >
    <Connect(Activity) />
  </Base>
</Base>
`;

exports[`AppVersion renders the details tab correctly 1`] = `
<Base
  paddingVertical="x10"
>
  <Base
    maxWidth={960}
  >
    <Tabs
      gap="x10"
    >
      <Tab
        active={true}
      >
        <Link
          as="some/path/details"
          href="/version?appSlug=app-slug-123&versionId=123&selectedTab=details"
        >
          <a>
            Details
          </a>
        </Link>
      </Tab>
      <Tab
        active={false}
      >
        <Link
          as="some/path/devices"
          href="/version?appSlug=app-slug-123&versionId=123&selectedTab=devices"
        >
          <a>
            Devices
          </a>
        </Link>
      </Tab>
      <Tab
        active={false}
      >
        <Link
          as="some/path/qa"
          href="/version?appSlug=app-slug-123&versionId=123&selectedTab=qa"
        >
          <a>
            Qa
          </a>
        </Link>
      </Tab>
      <Tab
        active={false}
      >
        <Link
          as="some/path/activity"
          href="/version?appSlug=app-slug-123&versionId=123&selectedTab=activity"
        >
          <a>
            Activity
          </a>
        </Link>
      </Tab>
    </Tabs>
  </Base>
  <Divider
    color="gray-2"
    direction="horizontal"
    width="0.125rem"
  />
  <Base
    maxWidth={960}
  >
    <Connect(AppVersionDetails) />
  </Base>
</Base>
`;

exports[`AppVersion renders the devices tab correctly 1`] = `
<Base
  paddingVertical="x10"
>
  <Base
    maxWidth={960}
  >
    <Tabs
      gap="x10"
    >
      <Tab
        active={false}
      >
        <Link
          as="some/path/details"
          href="/version?appSlug=app-slug-123&versionId=123&selectedTab=details"
        >
          <a>
            Details
          </a>
        </Link>
      </Tab>
      <Tab
        active={true}
      >
        <Link
          as="some/path/devices"
          href="/version?appSlug=app-slug-123&versionId=123&selectedTab=devices"
        >
          <a>
            Devices
          </a>
        </Link>
      </Tab>
      <Tab
        active={false}
      >
        <Link
          as="some/path/qa"
          href="/version?appSlug=app-slug-123&versionId=123&selectedTab=qa"
        >
          <a>
            Qa
          </a>
        </Link>
      </Tab>
      <Tab
        active={false}
      >
        <Link
          as="some/path/activity"
          href="/version?appSlug=app-slug-123&versionId=123&selectedTab=activity"
        >
          <a>
            Activity
          </a>
        </Link>
      </Tab>
    </Tabs>
  </Base>
  <Divider
    color="gray-2"
    direction="horizontal"
    width="0.125rem"
  />
  <Base
    maxWidth={960}
  >
    <Connect(AppVersionDevices) />
  </Base>
</Base>
`;

exports[`AppVersion renders the other tabs correctly 1`] = `
<Base
  paddingVertical="x10"
>
  <Base
    maxWidth={960}
  >
    <Tabs
      gap="x10"
    >
      <Tab
        active={false}
      >
        <Link
          as="some/path/details"
          href="/version?appSlug=app-slug-123&versionId=123&selectedTab=details"
        >
          <a>
            Details
          </a>
        </Link>
      </Tab>
      <Tab
        active={false}
      >
        <Link
          as="some/path/devices"
          href="/version?appSlug=app-slug-123&versionId=123&selectedTab=devices"
        >
          <a>
            Devices
          </a>
        </Link>
      </Tab>
      <Tab
        active={false}
      >
        <Link
          as="some/path/qa"
          href="/version?appSlug=app-slug-123&versionId=123&selectedTab=qa"
        >
          <a>
            Qa
          </a>
        </Link>
      </Tab>
      <Tab
        active={false}
      >
        <Link
          as="some/path/activity"
          href="/version?appSlug=app-slug-123&versionId=123&selectedTab=activity"
        >
          <a>
            Activity
          </a>
        </Link>
      </Tab>
    </Tabs>
  </Base>
  <Divider
    color="gray-2"
    direction="horizontal"
    width="0.125rem"
  />
  <Base
    maxWidth={960}
  >
    <h1>
      whateever
    </h1>
  </Base>
</Base>
`;

exports[`AppVersion renders the qa tab correctly 1`] = `
<Base
  paddingVertical="x10"
>
  <Base
    maxWidth={960}
  >
    <Tabs
      gap="x10"
    >
      <Tab
        active={false}
      >
        <Link
          as="some/path/details"
          href="/version?appSlug=app-slug-123&versionId=123&selectedTab=details"
        >
          <a>
            Details
          </a>
        </Link>
      </Tab>
      <Tab
        active={false}
      >
        <Link
          as="some/path/devices"
          href="/version?appSlug=app-slug-123&versionId=123&selectedTab=devices"
        >
          <a>
            Devices
          </a>
        </Link>
      </Tab>
      <Tab
        active={true}
      >
        <Link
          as="some/path/qa"
          href="/version?appSlug=app-slug-123&versionId=123&selectedTab=qa"
        >
          <a>
            Qa
          </a>
        </Link>
      </Tab>
      <Tab
        active={false}
      >
        <Link
          as="some/path/activity"
          href="/version?appSlug=app-slug-123&versionId=123&selectedTab=activity"
        >
          <a>
            Activity
          </a>
        </Link>
      </Tab>
    </Tabs>
  </Base>
  <Divider
    color="gray-2"
    direction="horizontal"
    width="0.125rem"
  />
  <Base
    maxWidth={960}
  >
    <_default />
  </Base>
</Base>
`;
